DSharpPlus.Caching/Memory/MemoryCacheService.cs

```diff
@@ -0,0 +1,91 @@
+using System;
+using System.Threading.Tasks;
+
+using DSharpPlus.Caching.Abstractions;
+
+using Microsoft.Extensions.Caching.Memory;
+using Microsoft.Extensions.Options;
+
+namespace DSharpPlus.Caching.Memory
+{
+    /// <summary>
+    /// An implementation of <see cref="ICacheService"/> relying on a memory cache.
+    /// </summary>
+    public class MemoryCacheService : ICacheService
```

Comment:
This is tech debt; the cache service should be a thin CRUD wrapper around the abstract *backing store*, and the backing store service is what handles the state of the data. 

For example in Remora:

[Implementation-agnostic cache service](https://github.com/Remora/Remora.Discord/blob/main/Backend/Remora.Discord.Caching/Services/CacheService.cs#L306-L311)

[IMemoryCache provider](https://github.com/Remora/Remora.Discord/blob/main/Backend/Remora.Discord.Rest/Caching/MemoryCacheProvider.cs#L51-L76)

[Redis provider](https://github.com/Remora/Remora.Discord/blob/main/Backend/Remora.Discord.Caching.Redis/Services/RedisCacheProvider.cs#L63-L87)